src/UbiHttp.cpp

```diff
@@ -244,100 +310,54 @@ float UbiHTTP::get(const char* device_label, const char* variable_label) {
  *         false if timeout is reached.
  */
 
-void UbiHTTP::reconnect(const char* host, const int port) {
+void UbiHTTP::reconnect(const char *host, const int port) {
   uint8_t attempts = 0;
   while (!_client_https_ubi.connected() && attempts < _maxReconnectAttempts) {
     if (_debug) {
-      Serial.print("Trying to connect to ");
+      Serial.print(F("Trying to connect to "));
       Serial.print(host);
-      Serial.print(" , attempt number: ");
+      Serial.print(F(" , attempt number: "));
       Serial.println(attempts);
     }
     _client_https_ubi.connect(host, port);
     if (_debug) {
-      Serial.println("Attempt finished");
+      Serial.println(F("Attempt finished"));
     }
     attempts += 1;
     delay(1000);
   }
 }
 
-float UbiHTTP::parseHttpAnswer(const char* request_type, char* data) {
-  float result = ERROR_VALUE;
-
-  // LV
-  if (request_type == "LV") {
-    char parsed[20];
-    char dst[20];
-    int len = strlen(data);  // Length of the answer char array from the server
-
-    for (int i = 0; i < len - 2; i++) {
-      if ((data[i] == '\r') && (data[i + 1] == '\n') && (data[i + 2] == '\r') && (data[i + 3] == '\n')) {
-        strncpy(parsed, data + i + 4, 20);  // Copies the result to the parsed
-        parsed[20] = '\0';
-        break;
-      }
-    }
-
-    /* Extracts the the value */
-
-    uint8_t index = 0;
-
-    // Creates pointers to split the value
-    char* pch = strchr(parsed, '\n');
-    if (pch == NULL) {
-      return result;
-    }
-
-    char* pch2 = strchr(pch + 1, '\n');
-
-    if (pch2 == NULL) {
-      return result;
-    }
-
-    index = (int)(pch2 - pch - 1);
-
-    sprintf(dst, "%s", pch);
-    dst[strlen(dst) - 1] = '\0';
-
-    float result = atof(dst);
-    return result;
-  }
-
-  return ERROR_VALUE;
-}
-
 /**
  * @arg response [Mandatory] Pointer to store the server's answer
  */
 
-void UbiHTTP::readServerAnswer(char* response) {
-  // Fills with zeros
-  for (int i = 0; i <= MAX_BUFFER_SIZE; i++) {
-    response[i] = '\0';
-  }
+void UbiHTTP::readServerAnswer(char *_serverResponse) {
 
-  if (_debug) {
-    Serial.println("----------");
-    Serial.println("Server's response:");
-  }
-
-  int j = 0;
+  sprintf(_serverResponse, "%c", _client_https_ubi.read());
 
   while (_client_https_ubi.available()) {
-    response[j] = (char)_client_https_ubi.read();
-    if (_debug) {
-      Serial.write(response[j]);
-    }
-    j++;
-    if (j >= MAX_BUFFER_SIZE - 1) {
+    char *c = (char *)malloc(sizeof(char) * 3);
```

Comment:
Add here as a comment, the expected server's response to further understand the parser that you have implemented